torchvision/transforms/functional.py

```diff
@@ -1234,6 +1234,9 @@ def affine(
     return F_t.affine(img, matrix=matrix, interpolation=interpolation.value, fill=fill)
 
 
+# Looks like to_grayscale() is a stand-alone functional that is never called
+# from the transform classes. Perhaps it's still here for BC? I can't be
+# bothered to dig. Anyway, this can be deprecated as we migrate to V2.
```

Comment:
IIUC, the difference is that `to_grayscale` is just for PIL images and thus supports arbitrary input color spaces or rather all that `Pillow` supports. Although there is no functional difference for PIL images in `rgb_to_grayscale`, tensors have to have the three RGB channels.

+1 on deprecating.